File: packages/java/src/main/java/org/imzala/UploadPartyInput.java
package org.imzala;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

/** One signing party for {@code DemandsResource.uploadDocument}. Email or phone (or both) required per party. */
@JsonPropertyOrder({"first_name", "last_name", "email", "phone"})
@JsonInclude(JsonInclude.Include.NON_NULL)
public final class UploadPartyInput {

  @JsonProperty("first_name")
  private final String firstName;

  @JsonProperty("last_name")
  private final String lastName;

  @JsonProperty("email")
  private final String email;

  /** E.164 format (e.g. {@code "+905551234567"}). */
  @JsonProperty("phone")
  private final String phone;

  public UploadPartyInput(String firstName, String lastName, String email, String phone) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.phone = phone;
  }

  public String getFirstName() {
    return firstName;
  }

  public String getLastName() {
    return lastName;
  }

  public String getEmail() {
    return email;
  }

  public String getPhone() {
    return phone;
  }
}

File: packages/php/src/UploadPartyInput.php
<?php

declare(strict_types=1);

namespace Imzala;

/** One signing party for {@see DemandsResource::uploadDocument}. Email or phone (or both) required per party. */
final class UploadPartyInput
{
    public function __construct(
        public readonly ?string $firstName = null,
        public readonly ?string $lastName = null,
        public readonly ?string $email = null,
        /** E.164 format (e.g. {@code "+905551234567"}). */
        public readonly ?string $phone = null,
    ) {
    }

    /** @return array<string, string> */
    public function toArray(): array
    {
        $result = [];
        if ($this->firstName !== null) {
            $result['first_name'] = $this->firstName;
        }
        if ($this->lastName !== null) {
            $result['last_name'] = $this->lastName;
        }
        if ($this->email !== null) {
            $result['email'] = $this->email;
        }
        if ($this->phone !== null) {
            $result['phone'] = $this->phone;
        }
        return $result;
    }
}
